syntax = "proto3";

import 'google/protobuf/timestamp.proto';

message SearchMessage {
    int64 person_id = 1;
    google.protobuf.Timestamp start_date = 2;
    google.protobuf.Timestamp end_date = 3;
    float meters = 4;
}

message PersonMessage {
    int64 id = 1;
    string first_name = 2;
    string last_name = 3;
    string company_name = 4;
}

message LocationMessage {
    int64 id = 1;
    int64 person_id = 2;
    string longitude = 3;
    string latitude = 4;
    google.protobuf.Timestamp creation_time = 5;
    string wkt_shape = 6;
}

message ConnectMessage {
    LocationMessage location = 1;
    PersonMessage person = 2;
}

message ConnectMessages {
    repeated ConnectMessage connections = 1;
}

service ConnectMessageService {
    rpc FindConnections(SearchMessage) returns (ConnectMessages);
}